Remotion/Core/Core/Context/ISafeContextStorageProvider.cs

```diff
@@ -24,6 +24,13 @@ namespace Remotion.Context
   // By default, use HttpContextStorageProvider. Only if not available, use CallContextStorageProvider.
   public interface ISafeContextStorageProvider
   {
+    /// <summary>
+    /// Opens a <see cref="SafeContextBoundary"/> into which values will not flow.
+    /// The previous context is restored when the <see cref="SafeContextBoundary"/> is disposed.
+    /// </summary>
+    /// <returns>A <see cref="SafeContextBoundary"/> object with which the previous context can be restored.</returns>
+    SafeContextBoundary OpenSafeContextBoundary ();
```

Comment:
The docs also imply that the boundary is unconditional, i.e. also works with CallContext and HttpContext.